cpp: warning: '= delete' with a message is a C++2c extension [-Wc++26-extensions]

### Bug Description

After updating to the current `master`, I get tons of C++ compiler warnings:

```
/build/release/libs/librepcb/ui/ui.h:51723:76: warning: '= delete' with a message is a C++2c extension [-Wc++26-extensions]
 51723 |         auto invoke_set_current_index (int) const = SLINT_DELETED_FUNCTION("the function 'set-current-index' is declared as private. Declare it as 'public'");
```

CMakeLists.txt:

```cmake
set(CMAKE_CXX_STANDARD 20)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)
```

### Reproducible Code (if applicable)

```slint

```

### Environment Details

- Slint Version: master
- Platform/OS: Linux
- Programming Language: C++
- Backend/Renderer: Qt

```
clang version 22.1.1
Target: x86_64-pc-linux-gnu
```

### Product Impact

_No response_

--- a/api/cpp/include/private/slint_config.h
+++ b/api/cpp/include/private/slint_config.h
@@ -25,7 +25,7 @@
 #    endif
 #endif // !defined(DOXYGEN)
 
-#if defined(__cpp_deleted_function) && __cpp_deleted_function >= 202403L
+#if defined(__cpp_deleted_function) && __cpp_deleted_function >= 202403L && __cplusplus > 202302L
 #    define SLINT_DELETED_FUNCTION(reason) delete (reason)
 #else
 #    define SLINT_DELETED_FUNCTION(reason) delete

